subsys/bluetooth/host_extensions/radio_notification_conn_cb.c

```diff
@@ -0,0 +1,189 @@
+#define EGU_INST_IDX 2
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU3)
+#define EGU_INST_IDX 3
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU4)
+#define EGU_INST_IDX 4
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU5)
+#define EGU_INST_IDX 5
+#elif defined(CONFIG_BT_RADIO_NOTIFICATION_CONN_CB_EGU_INST_EGU020)
+#define EGU_INST_IDX 020
+#else
+#error "Unknown EGU instance"
+#endif
+
+static nrfx_egu_t egu_inst = NRFX_EGU_INSTANCE(EGU_INST_IDX);
+static uint32_t configured_prepare_distance_us;
+static struct bt_radio_notification_conn_cb *registered_cb;
+
+static struct bt_conn *conn_pointers[CONFIG_BT_MAX_CONN];
+static uint8_t ppi_channels[CONFIG_BT_MAX_CONN];
+static struct k_work_delayable work_queue_items[CONFIG_BT_MAX_CONN];
+
+static uint32_t conn_interval_us_get(const struct bt_conn *conn)
+{
+	struct bt_conn_info info;
+
+	if (bt_conn_get_info(conn, &info)) {
+		return 0;
+	}
+
+	return BT_CONN_INTERVAL_TO_US(info.le.interval);
+}
+
+static void work_handler(struct k_work *work)
+{
+	uint8_t index = ARRAY_INDEX(work_queue_items, work);
+	struct bt_conn *conn = conn_pointers[index];
+
+	registered_cb->prepare(conn);
+
+	uint32_t conn_interval_us = conn_interval_us_get(conn);
+
+	/* Schedule the callback in the connection event does not occur because
+	 * of peripheral latency or scheduling conflicts.
+	 */
+	k_work_reschedule(&work_queue_items[index], K_USEC(conn_interval_us));
+}
+
+static void egu_isr_handler(uint8_t event_idx, void *context)
+{
+	ARG_UNUSED(context);
+
+	__ASSERT_NO_MSG(event_idx < ARRAY_SIZE(work_queue_items));
+
+	struct bt_conn *conn = conn_pointers[event_idx];
+	uint32_t conn_interval_us = conn_interval_us_get(conn);
+
+	__ASSERT_NO_MSG(conn_interval_us > configured_prepare_distance_us);
+
+	const uint32_t timer_distance_us =
+		conn_interval_us - configured_prepare_distance_us;
+
+	k_work_reschedule(&work_queue_items[event_idx], K_USEC(timer_distance_us));
+}
+
+static int setup_connection_event_trigger(struct bt_conn *conn)
+{
+	int err;
+	uint8_t conn_index;
+	uint16_t conn_handle;
+
+	err = bt_hci_get_conn_handle(conn, &conn_handle);
+	if (err) {
+		return err;
+	}
+
+	conn_index = bt_conn_index(conn);
+
+	sdc_hci_cmd_vs_get_next_conn_event_counter_t get_next_conn_event_count_params = {
+		.conn_handle = conn_handle,
+	};
+	sdc_hci_cmd_vs_get_next_conn_event_counter_return_t get_next_conn_event_count_ret_params;
+
+	err = hci_vs_sdc_get_next_conn_event_counter(
+		&get_next_conn_event_count_params, &get_next_conn_event_count_ret_params);
+	if (err) {
+		return err;
+	}
+
+	const sdc_hci_cmd_vs_set_conn_event_trigger_t params = {
+		.conn_handle = conn_handle,
+		.role = SDC_HCI_VS_CONN_EVENT_TRIGGER_ROLE_CONN,
+		.ppi_ch_id = ppi_channels[conn_index],
+		.period_in_events = 1, /* Trigger every connection event. */
+		.conn_evt_counter_start =
+			get_next_conn_event_count_ret_params.next_conn_event_counter + 10,
```

Comment:
+10 ? What does it mean or why is it needed?

Reply:
Added a comment in the code. We indent to remove this logic later